specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2020-08-13-aml-preview/storagecache.json

```diff
@@ -0,0 +1,2623 @@
+          "x-ms-enum": {
+            "name": "FirmwareStatusType",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "type": "string"
+        },
+        "firmwareUpdateDeadline": {
+          "description": "Time at which the pending firmware update will automatically be installed on the amlFilesystem.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "lastFirmwareUpdate": {
+          "description": "Time of the last successful firmware update.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "pendingFirmwareVersion": {
+          "description": "When firmwareUpdateAvailable is true, this field holds the version string for the update.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "readOnly" : true,
+      "type": "object"
+    },
+    "amlFilesystemHealth": {
+      "description": "An indication of amlFilesystem health. Gives more information about health than just that related to provisioning.",
+      "properties": {
+        "state": {
+          "description": "List of amlFilesystem health states.",
+          "enum": [
+            "Unknown",
+            "Healthy",
+            "Down",
+            "Transitioning",
+            "Maintenance"
+          ],
+          "x-ms-enum": {
+            "name": "amlFilesystemHealthStateType",
+            "modelAsString": true
+          },
+          "type": "string"
+        },
+        "statusDescription": {
+          "description": "Describes of state.",
+          "type": "string"
+        }
+      }
+    },
+    "amlFilesystemEncryptionSettings": {
+      "description": "amlFilesystem encryption settings.",
+      "properties": {
+        "keyEncryptionKey": {
+          "x-ms-secret": true,
+          "$ref": "#/definitions/KeyVaultKeyReference",
+          "description": "Specifies the location of the encryption key in Key Vault."
+        }
+      }
+    },
+    "amlFilesystemHSMSettings": {
+      "description": "amlFilesystem HSM settings.",
+      "properties": {
+        "container": {
+          "description": "Container used for the amlFilesystem.",
+          "$ref": "#/definitions/URLString"
+        },
+        "containerKeyVaultKey": {
+          "x-ms-secret": true,
+          "$ref": "#/definitions/KeyVaultKeyReference",
+          "description": "Specifies the location of the encryption key in Key Vault."
+        },
+        "importPrefix" : {
+          "description": "Path/Prefix for blobs in above container.  Only blobs containing this prefix get pre-hydrated into the cluster namespace",
+          "type" : "string"
+        },
+        "exportPrefix" : {
+          "description": "Path/Prefix in above blob container where data gets exported when an archive task runs",
+          "type" : "string"
+        }
+      }
+    },
+    "amlFilesystemHydrationStatus" : {
```

Comment:
This is specific to namespace hydration, so how about amlFilesystemNamespaceHydrationStatus? I know that's a mouthful.

Reply:
fixed, see on next commit.